server/svix-server/src/wait_for.rs

```diff
@@ -0,0 +1,78 @@
+use std::{net::IpAddr, time::Duration};
+
+use trust_dns_resolver::{
+    config::{ResolverConfig, ResolverOpts},
+    TokioAsyncResolver, TokioHandle,
+};
+
+use crate::cfg::Configuration;
+
+const POSTGRES_NAME: &str = "PostgreSQL";
+const POSTGRES_PORT: u16 = 5432;
+
+const REDIS_NAME: &str = "Redis";
+const REDIS_PORT: u16 = 6379;
+
+pub async fn wait_for(cfg: &Configuration, wait_for_seconds: u64) {
+    if let Some(redis_dsn) = &cfg.redis_dsn {
+        tokio::join!(
+            wait_for_inner(&cfg.db_dsn, POSTGRES_NAME, POSTGRES_PORT, wait_for_seconds),
+            wait_for_inner(redis_dsn, REDIS_NAME, REDIS_PORT, wait_for_seconds),
+        );
+    } else {
+        wait_for_inner(&cfg.db_dsn, POSTGRES_NAME, POSTGRES_PORT, wait_for_seconds).await
+    }
```

Comment:
Nitpick: this feels like application logic, rather than `wait_for` logic, so it feels like it should be in `main.rs`.